Microsoft.Toolkit.Uwp.UI.Controls/TextBoxMask/TextBoxMask.cs

```diff
@@ -0,0 +1,303 @@
+        private static readonly KeyValuePair<char, string> NumericCharacterRepresentation = new KeyValuePair<char, string>('9', "[0-9]");
+        private static readonly KeyValuePair<char, string> AlphaNumericRepresentation = new KeyValuePair<char, string>('*', "[A-Za-z0-9]");
+
+        private static void OnMaskChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void OnPlaceHolderChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void OnCustomMaskChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void InitTextBoxMask(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var textbox = d as TextBox;
+
+            // incase no value is provided us it as normal textbox
+            var mask = textbox?.GetValue(MaskProperty) as string;
+            if (string.IsNullOrWhiteSpace(mask))
+            {
+                return;
+            }
+
+            var placeHolderValue = textbox.GetValue(PlaceHolderProperty) as string;
+            if (string.IsNullOrEmpty(placeHolderValue))
+            {
+                return;
+            }
+
+            var placeHolder = placeHolderValue[0];
+
+            var representationDictionary = textbox.GetValue(RepresentationDictionaryProperty) as Dictionary<char, string>;
+            representationDictionary = new Dictionary<char, string>();
+            representationDictionary.Add(AlphaCharacterRepresentation.Key, AlphaCharacterRepresentation.Value);
+            representationDictionary.Add(NumericCharacterRepresentation.Key, NumericCharacterRepresentation.Value);
+            representationDictionary.Add(AlphaNumericRepresentation.Key, AlphaNumericRepresentation.Value);
+
+            var customDictionaryValue = textbox.GetValue(CustomMaskProperty) as string;
+            if (!string.IsNullOrWhiteSpace(customDictionaryValue))
+            {
+                var customRoles = customDictionaryValue.Split(',');
+                foreach (var role in customRoles)
+                {
+                    var roleValues = role.Split(':');
+                    if (roleValues.Length != 2)
+                    {
+                        throw new Exception("Invalid custom mask");
+                    }
+
+                    var keyValue = roleValues[0];
+                    var value = roleValues[1];
+                    char key;
+
+                    // an exception should be throw if the regex is not valid
+                    Regex.Match(string.Empty, value);
+                    if (!char.TryParse(keyValue, out key))
+                    {
+                        throw new Exception("Invalid custom mask, please validate the mask key");
```

Comment:
Should we throw an InvalidOperationException (or something similar) here instead of a basic Exception?